subiquity/server/controllers/ad.py

```diff
@@ -34,6 +39,92 @@ async def GET(self) -> Optional[ADConnectionInfo]:
         return self.model.conn_info
 
     async def POST(self, data: ADConnectionInfo) -> List[ADValidationResult]:
+        """ Configures this controller with the supplied info.
+            Returns a list of errors if the info submitted is invalid or
+            [ADValidationResult.OK] on success. """
+        result = check_domain_name(data.domain_name) | \
```

Comment:
style nitpick:  I might structure it as
```
result = set()
result |= check_a()
result |= check_b()
```
Also what was done in check_domain_name is fine.
That should make for cleaner diffs if this is tweaked in the future, and I feel like validation rules are a high traffic bit of code anywhere they are found.